File: src/main/java/dev/sky_lock/pocketlifevehicle/vehicle/CarEntity.java
package dev.sky_lock.pocketlifevehicle.vehicle;

import org.bukkit.Location;

import java.util.UUID;

/**
 * @author sky_lock
 */

public class CarEntity {
    private final String ownerUuid;
    private final String modelId;
    private final Location location;
    private final float fuel;

    CarEntity(String ownerUuid, String modelId, Location location, float fuel) {
        this.ownerUuid = ownerUuid;
        this.modelId = modelId;
        this.location = location;
        this.fuel = fuel;
    }

    UUID getOwner() {
        return UUID.fromString(ownerUuid);
    }

    String getModelId() {
        return modelId;
    }

    public Location getLocation() {
        return location;
    }

    public float getFuel() {
        return fuel;
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/vehicle/CarEntity.kt
package dev.sky_lock.pocketlifevehicle.vehicle

import org.bukkit.Location
import java.util.*

/**
 * @author sky_lock
 */
class CarEntity internal constructor(private val ownerUuid: String, val modelId: String, val location: Location, val fuel: Float) {
    val owner: UUID
        get() = UUID.fromString(ownerUuid)

}